File: python/pi.py
from time import time
import math


def calculate_pi(time_allowed):
    time_start = time()
    counter = 1
    total_sum = 0
    while time_start + time_allowed > time():
        total_sum += 1 / counter ** 2
        counter += 1
    return math.sqrt(total_sum * 6)


if __name__ == '__main__':
    print(calculate_pi(5))

File: java/pi.java
class getpiintime{
    public static void main(String args[]){
        System.out.println(calculatePi(5));
    }

    private static double calculatePi(int timeAllowed){
        int counter = 1;
        double totalSum = 0;
        long start = System.currentTimeMillis();
        while ((System.currentTimeMillis()-start)/1000F < 5){
            totalSum += 1.0/Math.pow(counter, 2);
            counter += 1;
        }
        return Math.sqrt(totalSum*6);
    }
}


class getpiinperiod{
    public static void main(String args[])
    {
        double place = 0;
        double k = 0;
        double result = 0;
        while(true)
        {
            result = method(k,result);
            k++;
            if(place==10000000) System.out.println(result * 4.0);;
        }
    }
    public static double method(double k, double num)
    {
        return (Math.pow(-1.0,k) / (2.0*k + 1.0) + num);
    }
}
